### PR TITLE
Correct JSON format for special float values

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally 
- [x] New functions marked with `` `r since("VERSION")` ``
- [x] Declare copyright holder and open-source license: see below

#### Summary
See https://github.com/stan-dev/stan/issues/3142#issuecomment-1256219024  or [rapidjson doc](https://rapidjson.org/namespacerapidjson.html#a81379eb4e94a0386d71d15fda882ebc9a857894ab51cafa62956e9c9f0dafc4d8). The currently recommended values of `"inf"` and `"-inf"` are not only not the only options, they actually fail to parse entirely. 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

File: src/cmdstan-guide/json_apdx.Rmd [hunk 1]
```diff
@@ -44,9 +44,12 @@ There is no distincition between integer and real numbers
 in the JSON format other than whether they have periods or scientific
 notation.
 
-- Positive infinity is represented as the string `"+inf"`,
-negative infinity as `"-inf"`, and not a number as `"NaN"`.  These
-strings may be mixed with numerical types.
+- The special floating point values for positive infinity, negative infinity, 
+and not-a-number can be represented in multiple ways.  Positive infinity can 
+represented as the string `"Inf"`, the string `"Infinity"`, or the atom `Infinity`.
```
Comment (bob-carpenter) [inline src/cmdstan-guide/json_apdx.Rmd:49]: can represented -> can be represented

File: src/cmdstan-guide/json_apdx.Rmd [hunk 2]
```diff
@@ -44,9 +44,12 @@ There is no distincition between integer and real numbers
...
+Negative infinity can represented as the string `"-Inf"`, the string `"-Infinity"`,
```
Comment (bob-carpenter) [inline src/cmdstan-guide/json_apdx.Rmd:50]: Same here.

File: src/cmdstan-guide/json_apdx.Rmd [hunk 3]
```diff
@@ -44,9 +44,12 @@ There is no distincition between integer and real numbers
...
 notation.
 
-- Positive infinity is represented as the string `"+inf"`,
-negative infinity as `"-inf"`, and not a number as `"NaN"`.  These
-strings may be mixed with numerical types.
+- The special floating point values for positive infinity, negative infinity, 
+and not-a-number can be represented in multiple ways.  Positive infinity can 
+represented as the string `"Inf"`, the string `"Infinity"`, or the atom `Infinity`.
+Negative infinity can represented as the string `"-Inf"`, the string `"-Infinity"`, 
+or the atom `-Infinity`. Not-a-number can be represented as as the string `"NaN"`
```
Comment (bob-carpenter) [inline src/cmdstan-guide/json_apdx.Rmd:51]: as as -> as

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
Typos.

**Comment by WardBrian** [comment]
Thanks for catching those @bob-carpenter - wrote too hastily in the web editor

**Review by bob-carpenter — APPROVED** [review]
thanks for fixing.

**Pull request merged**